main.tf

```diff
@@ -254,6 +254,8 @@ resource "null_resource" "bootstrap" {
     dcos_zk_agent_credentials                    = "${var.dcos_zk_agent_credentials}"
   }
 
+  count = "${var.enable_bootstrap ? 1 : 0}"
```

Comment:
why should I invoke this module without executing? Is this meant to be a fake conditional module? In that case every resource should have a count

Reply:
a length check on bootstrap_ip should do the same without having a new variable

Reply:
this will stop the null_resource from running on the remote region when there doesn't exist a remote bootstrap node there but a local region bootstrap node exist. Without this I noticed that when only one bootstrap node exist and there are two `resource "null_resource" "bootstrap"`, it will try to two bootstrap ssh commands on the same machine.

Reply:
I see but in that case `enable_bootstrap` doesn't sound right to me.

Reply:
This was a variable that is passed through from the convenience wrapper:

`enable_bootstrap`. I don't mind naming it something else, I just wanted to keep the meaning intact.

```
module "remote-1" {
  source = "dcos-terraform/remote-region/aws"
  version = "~> 0.1.0"

  providers = {
    aws = "aws.spoke-1"
  }
  dcos_instance_os     = "coreos_1855.5.0"
  cluster_name         = "region2-cluster2"
  ssh_public_key_file  = "~/.ssh/id_rsa.pub"
  dcos_version         = "1.12.1"
  admin_ips            = ["${data.http.whatismyip.body}/32"]

  # enable_bootstrap     = true
  master_private_ips     = "${module.dcos.infrastructure-masters.private_ips}"

  enable_bootstrap     = false
  bootstrap_private_ip = "${module.dcos.infrastructure-bootstrap.private_ip}"

  # review with SRE
  # bootstrap_prereq-id  = "${module.dcos.infrastructure-bootstrap.prereq-id}"
  # masters_prereq-id  = "${module.dcos.infrastructure-masters.prereq-id}"

  num_private_agents = "3"
  num_public_agents  = "1"
  subnet_range       = "172.13.0.0/16"

  dcos_variant              = "ee"
  dcos_license_key_contents = "${file("./license.txt")}"

  tags={owner = "mbernadin", expiration = "6h"}
  dcos_install_mode  = "${var.dcos_install_mode}"
}
```

Reply:
@geekbass and @fatz, this PR LGTM. Let me know if there is anything else needed here based on the last comment on `enable_bootstrap`. This name goes all the way up to the convenience wrapper and from the user, it seems like it would make sense. If not, you can let me know what variable name we should pick.

Reply:
@bernadinm I am okay with the variable here. I think it makes sense after testing the Mutli Region multiple times.

Reply:
thanks @geekbass for reviewing!